### PR TITLE
Small emulation enhancements

### PR DESCRIPTION
**Pull request by nbren12**
These are some small quality of life improvements and clean ups.

Delete a redundant class (TransformedModelConfig) and kill the training if the loss value is NaN. See commits for more motivation.

### CODE REVIEW

**Comment by nbren12** [comment]
Switched to using a tensorflow callback to kill the training. Here is what the output looks like:
```
Epoch 1/2
Batch 1: Invalid loss, terminating training
2/2 - 2s - loss: nan - cloud_water_mixing_ratio_after_precpd_loss: nan - specific_humidity_after_precpd_loss: nan - total_precipitation_loss: nan - air_temperature_after_precpd_loss: nan
```

**Review by AnnaKwa — APPROVED** [review]
LGTM!
